rest/class-boldgrid-backup-rest-archive.php

```diff
@@ -0,0 +1,277 @@
+							return (int) $field;
+						},
+					],
+				],
+			],
+			'schema' => [ $this, 'get_schema' ],
+		] );
+	}
+
+	/**
+	 * Register the route for viewing a list of backups.
+	 *
+	 * @since X.X.X
+	 */
+	public function register_list() {
+		register_rest_route( $this->namespace, '/' . $this->resource, [
+			[
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'get_items' ],
+				'permission_callback' => [ $this, 'permission_check' ],
+			],
+			'schema' => [ $this, 'get_schema' ],
+		] );
+	}
+
+	/**
+	 * Get our sample schema for an archive.
+	 *
+	 * @since X.X.X
+	 *
+	 * @return array Schema Format.
+	 */
+	public function get_schema() {
+		$schema = [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => $this->resource,
+			'type'       => 'object',
+			'properties' => [
+				'id'            => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Unique identifier for the object.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'title'         => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Name of the archive.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'description'   => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Description of the archive.', 'boldgrid-backup' ),
+					'type'        => 'array',
+				],
+				'filename'      => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Filename of backup.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'url'           => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Download link for the backup.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'creation_date' => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Date the archive was created.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+			],
+		];
+
+		return $schema;
+	}
+
+	/**
+	 * Prepare the item for the REST response.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param mixed $item WordPress representation of the item.
+	 * @param WP_REST_Request $request Request object.
+	 * @return mixed
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		return $this->filter_schema_properties( $item );
+	}
+
+	/**
+	 * Create a new archive.
+	 *
+	 * For examples, please see rest/README.MD
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @see Boldgrid_Backup_Rest_Controller::permission_check for more information if you are getting
+	 *                                                        a 401 unauthorized error.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return array                   Job Resource.
+	 */
+	public function create_item( $request ) {
+		// Initialize a new task.
+		$task = new Boldgrid_Backup_Admin_Task();
+		$task->init( [ 'type' => 'backup' ] );
```

Comment:
Arrays should be declared with long array syntax per WordPress coding standards
https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays

Reply:
See similar comment above.